src/_Raspberry_Pi_Driver.c

```diff
@@ -3,52 +3,58 @@
 #include <unistd.h>
 
 #include "erl_common.c"
-#include "erl_interface.h"
 #include "ei.h"
 
 #include "Raspberry_Pi/pi_sensor.h"
 
 typedef unsigned char byte;
 
-int main(int argc, char *argv[]) {
+int main(int argc, char *argv[])
+{
 
-  if (argc != 3) {
+  if (argc != 3)
+  {
     fprintf(stderr, "Usage: %s [GPIO_ECHO] [GPIO_TRIG] \n", argv[0]);
     exit(EXIT_FAILURE);
   }
 
+  ei_init();
+
   int echo = atoi(argv[1]);
   int trig = atoi(argv[2]);
 
-  ETERM * arr[4], *tuple;
-  unsigned char buf[BUFSIZ];
-
-  erl_init(NULL, 0);
-
-  while (read_cmd(buf) > 0) {
+  byte input[100];
+  ei_x_buff output;
 
+  while (read_cmd(input) > 0)
+  {
     float distance = 0;
     int result = pi_sensor(echo, trig, &distance);
 
-    if(result == SUCCESS){
-      arr[0] = erl_mk_atom("ok");
-      arr[1] = erl_mk_int(echo);
-      arr[2] = erl_mk_int(trig);
-      arr[3] = erl_mk_float(distance);
-    } else {
-      arr[0] = erl_mk_atom("error");
-      arr[1] = erl_mk_int(echo);
-      arr[2] = erl_mk_int(trig);
-      arr[3] = erl_mk_int(result);
-    }
+    ei_x_new(&output);
 
-    tuple  = erl_mk_tuple(arr, 4);
-
-    erl_encode(tuple, buf);
+    if (result == SUCCESS)
+    {
+      ei_x_format(&output,
+                  "[{~a,~i,~i,~f}]",
```

Comment:
I am curious, why we need the array here? why not only `{~a,~i,~i,~f}`

Reply:
Because I have no idea of C programming and just trial&errored my way into a working version.
I've removed the list in the latest commit.